File: Pila_Lista/PilaLista.js
const readline = require("readline");

class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

class Stack {
    constructor() { this.top = null; }

    push(item) {
        const n = new Node(item);
        n.next = this.top;
        this.top = n;
        console.log(`Elemento ${item} insertado.`);
    }
    pop() {
        if (!this.top) return console.log("Stack Underflow");
        const val = this.top.data;
        this.top = this.top.next;
        return val;
    }
    peek() {
        if (!this.top) return console.log("Pila vacía");
        return this.top.data;
    }
    isEmpty() { return this.top === null; }
}

const pila = new Stack();
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

function menu() {
    console.log("\n--- MENU PILA (JS Linked List) ---");
    console.log("1. Push\n2. Pop\n3. Peek\n4. isEmpty\n5. Salir");
    rl.question("Elige: ", op => {
        switch(parseInt(op)) {
            case 1: rl.question("Ingresa un valor: ", v=>{ pila.push(parseInt(v)); menu(); }); break;
            case 2: console.log("Eliminado:", pila.pop()); menu(); break;
            case 3: console.log("Tope:", pila.peek()); menu(); break;
            case 4: console.log(pila.isEmpty()? "Pila vacía":"No está vacía"); menu(); break;
            case 5: console.log("Saliendo..."); rl.close(); break;
            default: console.log("Opción inválida"); menu(); break;
        }
    });
}
menu();

File: Pila_Lista/PilaLista.cs
using System;

class Node {
    public int data;
    public Node next;
    public Node(int d) { data = d; }
}

class StackLinked {
    Node top = null;

    void Push(int item) {
        Node n = new Node(item);
        n.next = top;
        top = n;
        Console.WriteLine($"Elemento {item} insertado.");
    }

    int Pop() {
        if (top == null) { Console.WriteLine("Stack Underflow"); return -1; }
        int value = top.data;
        top = top.next;
        return value;
    }

    int Peek() {
        if (top == null) { Console.WriteLine("Pila vacía"); return -1; }
        return top.data;
    }

    bool IsEmpty() => top == null;

    static void Main() {
        StackLinked pila = new StackLinked();
        int opcion, valor;

        do {
            Console.WriteLine("\n--- MENU PILA (C# Linked List) ---");
            Console.WriteLine("1. Push\n2. Pop\n3. Peek\n4. isEmpty\n5. Salir");
            Console.Write("Elige: ");
            opcion = int.Parse(Console.ReadLine());

            switch(opcion) {
                case 1: Console.Write("Ingresa un valor: "); valor = int.Parse(Console.ReadLine()); pila.Push(valor); break;
                case 2: valor = pila.Pop(); if(valor!=-1) Console.WriteLine("Eliminado: "+valor); break;
                case 3: valor = pila.Peek(); if(valor!=-1) Console.WriteLine("Tope: "+valor); break;
                case 4: Console.WriteLine(pila.IsEmpty()? "Pila vacía":"No está vacía"); break;
                case 5: Console.WriteLine("Saliendo..."); break;
                default: Console.WriteLine("Opción inválida"); break;
            }
        } while(opcion != 5);
    }
}
